tests/by-util/test_expr.rs

```diff
@@ -364,3 +364,17 @@ fn test_eager_evaluation() {
         .fails()
         .stderr_contains("division by zero");
 }
+
+#[test]
+fn test_long_input() {
+    const MAX_NUMBER: usize = 40000;
```

Comment:
```suggestion
    // Giving expr an arbitrary long expression should succeed rather than end with a segfault due to a stack overflow.
    const MAX_NUMBER: usize = 40000;
```

I would add this comment, so people reading the code know why it exists.

Reply:
Ah yes good point I'll add this.